rsyslog/undefined_field/undefined_field.go

```diff
@@ -214,7 +214,7 @@ func processUndefinedAndMaxNumFields(input map[string]interface{}) (string, map[
 		for field := range undefMap {
 			delete(input, field)
 		}
-		if (checkMaxNumFields && count < 0) {
+		if (checkMaxNumFields && count < 0) || (cfg.UseUndefined && cfg.UndefinedToString) {
```

Comment:
This isn't the code that does the undefined-to-string processing for individual fields - it does the undefined-to-string processing for all of the undefined fields as a single JSON string of the undefined map value.

Reply:
I'm trying to solve this problem [1].  I think this configuration is supposed to do what you described...  Please note that _CDM_UNDEFINED_TO_STRING=true && CDM_USE_UNDEFINED=false_ case is working just fine [2].   Well, after my opening bz1735530, I found what I thought broken were all working as expected, and rather some other cases were not... :p

https://bugzilla.redhat.com/show_bug.cgi?id=1735530#c1
[1] - BAD_1) CDM_UNDEFINED_TO_STRING with CDM_USE_UNDEFINED=true
[2] - GOOD_3) CDM_UNDEFINED_TO_STRING

Reply:
ok - I updated the bz

Reply:
Thanks, @richm.  I reverted the change.